<a href="https://colab.research.google.com/github/azzamasim/titanicml/blob/main/Utterubbish_P1M8.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from sklearn.model_selection import train_test_split
import tensorflow as tf
import pandas as pd
import numpy as np

Importing data and pre-processing

In [2]:
df = pd.read_csv("https://cdn.techroulette.xyz/projects/shipwreck/data/m3_data.csv")

df = df.drop(['PassengerId', 'Name', 'Ticket', 'Cabin'], axis=1)
df = df.replace({'male': 0, 'female': 1})
df = df.replace({'S': 0, 'Q': 1, 'C': 2})

Creating training and testing data sets

In [3]:
y = df.pop('Survived')
x = df
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.3)

Building Model

In [4]:
model = tf.keras.models.Sequential()
model.add(tf.keras.layers.Dense(10, input_shape=(7,), activation='relu'))
model.add(tf.keras.layers.Dense(1, activation='sigmoid'))

Implementing ways to improve accuracy


In [5]:
#Step 1: imputing values (mostly for age)
from sklearn.impute import SimpleImputer

#Imputation
my_imputer = SimpleImputer()
imputed_x_train = pd.DataFrame(my_imputer.fit_transform(x_train))
imputed_x_test = pd.DataFrame(my_imputer.transform(x_test))
imputed_x_train.columns = x_train.columns
imputed_x_test.columns = x_test.columns

Training Model

In [6]:
model.compile(
    optimizer="adam",
    loss="binary_crossentropy", 
    metrics=["accuracy"]
  )

#model.compile(optimizer ="rmsprop", loss ="categorical_crossentropy", metrics =['accuracy'])

model.fit(imputed_x_train, y_train, epochs=125)

Epoch 1/125
20/20 [==============================] - 2s 4ms/step - loss: 2.8374 - accuracy: 0.5666
Epoch 2/125
20/20 [==============================] - 0s 4ms/step - loss: 1.5961 - accuracy: 0.5297
Epoch 3/125
20/20 [==============================] - 0s 7ms/step - loss: 1.0434 - accuracy: 0.6677
Epoch 4/125
20/20 [==============================] - 0s 21ms/step - loss: 0.9587 - accuracy: 0.6838
Epoch 5/125
20/20 [==============================] - 0s 10ms/step - loss: 0.8837 - accuracy: 0.6854
Epoch 6/125
20/20 [==============================] - 0s 8ms/step - loss: 0.8176 - accuracy: 0.6870
Epoch 7/125
20/20 [==============================] - 0s 7ms/step - loss: 0.7601 - accuracy: 0.6838
Epoch 8/125
20/20 [==============================] - 0s 5ms/step - loss: 0.6995 - accuracy: 0.6886
Epoch 9/125
20/20 [==============================] - 0s 7ms/step - loss: 0.6514 - accuracy: 0.6950
Epoch 10/125
20/20 [==============================] - 0s 24ms/step - loss: 0.6176 - accuracy: 0.7014
Epoch 

Testing Model

In [7]:
model.evaluate(imputed_x_test, y_test, verbose=2)

9/9 - 0s - loss: 0.4612 - accuracy: 0.7910 - 222ms/epoch - 25ms/step


[0.4611590504646301, 0.7910447716712952]

Tracking Results:

PassengerId, Name, Ticket, Cabin always dropped

Sex, Age, Pclass: Always included

Variables to be shuffled around: SibSp, Parch, Fare, Embarked, Pclass, 
1.   With Pclass, SibSp, Parch: **0.60 to 0.66**
2.   With Pclass, SibSp, Parch, Fare: **0.59 to 0.63**
3.   With all : **0.60 to 0.65**
4.   With Pclass, Fare, Embarked: **0.57 to 0.65**
5.   All with 200 epochs: **0.62 to 0.66**
6. After Imputation: **0.73 - 0.83**

